ignition-server/controllers/tokensecret_controller.go

```diff
@@ -0,0 +1,95 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	finalizer             = "hypershift.openshift.io/finalizer"
+	tokenSecretReleaseKey = "release"
+	tokenSecretTokenKey   = "token"
+	tokenSecretAnnotation = "hypershift.io/ignition-config"
+)
+
+// PayloadsStore contains an updated map of token: payload.
+var PayloadStore map[string][]byte
```

Comment:
This map needs guarded for concurrency all over the place, and represents state indirectly shared across thread boundaries.

Although you could use a global mutex or something, my first thought was to introduce one additional small abstraction for the payload cache itself. For example, both the controller and the HTTP handler could receive an `IgnitionPayloadCache` or something that encapsulates the map. The controller would handle calling `AddOrUpdate()` and `Delete()` calls on the cache, and the HTTP handler would make `Get()` calls, and internally the cache can implement thread safety behind those calls.

Reply:
done